### DISCUSSION TITLE
The best way to create a dynamic grid

### DISCUSSION THREAD

**Discussion by argolian** [filamentphp/filament#9466] [Help]
I'm trying to create an admin panel using filament. So far, so good, but have a hit a problem which I cannot find a suitable answer to:

I need a grid/input or something  which allows me to add the following as an example
// --------------------------------------------------------
 // |  Name of Holiday    |   Start Date    | End Date     |  <--- Just a header
 // -------------------------------------------------------
 // | Christmas Period    |  22-12-2023     | 04-01-2024   | <--- Links to a field in the database
 // | Easter Period       |  01-04-2024     | 07-04-2023   |  \|/ and so on
 // and so on 

 // New rows should only be added if the previous row has been completed
 // Fields are assigned as nullable in the database and there can be any number of
// these periods held within the database.

At the moment I have 50 rows but I'm sure this isn't the best methodology 

What's the best way to achieve this? All I've managed to find thus far is plenty of examples of extracting the data from a table but I want the option to input the data as this is designed to fit on a new record form.

Any suggestions appreciated.

**Comment by leandrocfe** [comment]
Maybe a [table repeater](https://filamentphp.com/plugins/icetalker-table-repeater) ?
